core/src/main/scala/org/http4s/Message.scala

```diff
@@ -267,6 +267,28 @@ final class Request[F[_]](
 
   def queryString: String = uri.query.renderString
 
+  /** cURL representation of the request.
+    *
+    * Supported Parameters are:
+    *
+    * -X,
+    * -H (redacts sensitive information such as Authorization- or Cookie-Headers)
+    */
+  def asCurl: String = {
+    val elements = List(
+      s"-X ${method.name}",
+      s"'${uri.renderString}'",
+      headers
+        .redactSensitive()
+        .toList
+        .map { header =>
+          s"-H '${header.toString}'"
```

Comment:
oh you mean regular quotation marks, like " ?

Reply:
You would need to replace every `'` in the value with with `'\''`: the first character closes it, the second character escapes it, and the third character reopens it.  I _think_ that would be sufficient.

---

If someone comes along and tells us that's wrong, Emacs' restclient takes a more paranoid approach.  Instead of quoting, it escapes anything but a handful of known safe characters:

```http
GET https://httpbin.org/get
X-escape-test: "Don't forget to escape!"
```

yields

```sh
curl -i -H X-escape-test\:\ \"Don\'t\ forget\ to\ escape\!\" -XGET https\://httpbin.org/get
```

The relevant code that does it is:

```emacs-lisp
      ;; Quote everything except POSIX filename characters.
      ;; This should be safe enough even for really weird shells.
      (replace-regexp-in-string
       "\n" "'\n'"
       (replace-regexp-in-string "[^-0-9a-zA-Z_./\n]" "\\\\\\&" argument))))
```